edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/controllers/DeviceScopeController.cs

```diff
@@ -32,102 +33,167 @@ public async Task GetDevicesAndModulesInTargetDeviceScope([FromRoute] string act
+        bool TryGetTargetDeviceId(string authChain, out string targetDeviceId)
         {
             targetDeviceId = string.Empty;
+
+            // The target device is the first ID in the provided authchain,
+            // which could be a module ID of the format "deviceId/moduleId".
             var actorAuthChainIds = authChain.Split(';', StringSplitOptions.RemoveEmptyEntries);
 
-            // Should have at least 1 element in the chain
-            if (actorAuthChainIds.Length < 1)
+            if (actorAuthChainIds.Length > 0)
             {
+                var ids = actorAuthChainIds[0].Split('/', StringSplitOptions.RemoveEmptyEntries);
+
+                if (ids.Length > 0)
+                {
+                    targetDeviceId = ids[0];
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        async Task<bool> AuthorizeActor(IEdgeHub edgeHub, string actorDeviceId, string actorModuleId, string targetId)
+        {
+            if (actorModuleId != Constants.EdgeHubModuleId)
+            {
+                // Only child EdgeHubs are allowed to act OnBehalfOf of devices/modules.
                 return false;
             }
 
-            // The actor device is the one that sent the request, so the last element
-            // of the auth-chain should always be the actor device's ID
-            if (!actorDeviceId.Equals(actorAuthChainIds.LastOrDefault()))
+            // Actor device is claiming to be our child, and that the target device is its child.
+            // So we should have an authchain already cached for the target device.
+            Option<string> targetAuthChain = await edgeHub.GetAuthChainForIdentity(targetId);
+
+            if (!targetAuthChain.HasValue)
             {
                 return false;
             }
 
-            // The target device ID is the first device ID in the chain, which could be a module ID
-            // of the format "deviceId/moduleId"
-            var ids = actorAuthChainIds[0].Split('/', StringSplitOptions.RemoveEmptyEntries).ToList();
-            targetDeviceId = ids[0];
+            // Validate the the target auth-chain
+            if (!AuthChainHelpers.ValidateAuthChain(actorDeviceId, targetId, targetAuthChain.Expect(() => new InvalidOperationException())))
+            {
+                return false;
+            }
 
             return true;
         }
 
+        async Task SendResponse(EdgeHubScopeResult result)
+        {
+            this.Response.StatusCode = (int)result.Status;
+
+            if (result.Status == HttpStatusCode.OK)
+            {
+                var resultJsonContent = JsonConvert.SerializeObject(result);
+                var resultUtf8Bytes = Encoding.UTF8.GetBytes(resultJsonContent);
+
+                this.Response.ContentLength = resultUtf8Bytes.Length;
+                this.Response.ContentType = SupportedContentType;
```

Comment:
```suggestion
                this.Response.ContentType = ApplicationJsonContentType;
```

Reply:
where is ApplicationJsonContentType defined?

---
In reply to: [458524456](https://github.com/Azure/iotedge/pull/3248#discussion_r458524456) [](ancestors = 458524456)